vcr/issuer/issuer.go

```diff
@@ -139,9 +140,58 @@ func (i issuer) buildVC(credentialOptions vc.VerifiableCredential) (*vc.Verifiab
 	return signedCredential, nil
 }
 
-func (i issuer) Revoke(credentialID ssi.URI) error {
-	//TODO implement me
-	panic("implement me")
+func (i issuer) Revoke(credentialID ssi.URI) (*credential.Revocation, error) {
+	// first find it using a query on id.
+	credentialToRevoke, err := i.store.GetCredential(credentialID)
+	if err != nil {
+		return nil, fmt.Errorf("could not revoke: %w", err)
+	}
+
+	revocation, err := i.buildRevocation(*credentialToRevoke)
+	if err != nil {
+		return nil, err
+	}
+
+	err = i.publisher.PublishRevocation(*revocation)
+	if err != nil {
+		return nil, fmt.Errorf("failed to publish revocation: %w", err)
+	}
+
+	log.Logger().Infof("Verifiable Credential revoked (id=%s)", credentialToRevoke.ID)
+	return revocation, nil
+}
+
+func (i issuer) buildRevocation(credentialToRevoke vc.VerifiableCredential) (*credential.Revocation, error) {
+	// find issuer
+	issuerDID, err := did.ParseDID(credentialToRevoke.Issuer.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract issuer: %w", err)
+	}
+
+	assertionKey, err := i.keyResolver.ResolveAssertionKey(*issuerDID)
+	// set defaults
+	revocation := credential.BuildRevocation(credentialToRevoke)
+
+	revocationAsMap := map[string]interface{}{}
+	b, _ := json.Marshal(revocation)
+	_ = json.Unmarshal(b, &revocationAsMap)
+
+	ldProof := proof.NewLDProof(proof.ProofOptions{Created: time.Now()})
+	signingResult, err := ldProof.Sign(revocationAsMap, signature.JSONWebSignature2020{ContextLoader: i.contextLoader}, assertionKey)
+	if err != nil {
+		return nil, err
+	}
+
+	signingResultAsMap, ok := signingResult.(proof.SignedDocument)
+	if !ok {
+		return nil, errors.New("unable to cast signing result to SignedDocument")
```

Comment:
`LDProof.Sign()` always returns a `SignedDocument`, and otherwise it isn't a recoverable error (by retrying/modifying input). So I'd say don't check the cast and let it blow up if it fails.

Reply:
Will result in nasty stacktraces, but that is probably a good thing.